component/service/src/main/java/org/exoplatform/social/service/rest/IdentitySearchRESTService.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2003-2020 eXo Platform SAS.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see<http://www.gnu.org/licenses/>.
+ */
+package org.exoplatform.social.service.rest;
+
+import java.nio.file.AccessDeniedException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriInfo;
+
+import org.exoplatform.commons.utils.ListAccess;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.Group;
+import org.exoplatform.services.organization.OrganizationService;
+import org.exoplatform.services.rest.resource.ResourceContainer;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.profile.ProfileFilter;
+import org.exoplatform.social.core.service.LinkProvider;
+import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.core.space.spi.SpaceService;
+import org.exoplatform.social.service.rest.editors.ErrorMessage;
+import org.exoplatform.social.service.rest.editors.HypermediaLink;
+import org.exoplatform.social.service.rest.editors.HypermediaSupport;
+import org.exoplatform.social.service.rest.editors.IdentityData;
+
+/**
+ * The Class IdentitySearchRESTService.
+ */
+@Path("/identity/search")
+public class IdentitySearchRESTService implements ResourceContainer {
```

Comment:
Why adding a new endpoint for users ?
Isn't sufficient the following endpoints:
* https://github.com/Meeds-io/social/blob/develop/component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java
* https://github.com/Meeds-io/social/blob/develop/component/service/src/main/java/org/exoplatform/social/rest/impl/userrelationship/UsersRelationshipsRestResourcesV1.java
* https://github.com/Meeds-io/social/blob/develop/component/service/src/main/java/org/exoplatform/social/rest/impl/relationship/RelationshipsRestResourcesV1.java

If it's not, I suggest to enrich those existing.

What is the main idea behing the following endpoints ? is it for suggestion ? we have other endpoints for this too. I invite you to know more about it : 
* https://github.com/Meeds-io/social/blob/develop/component/service/src/main/java/org/exoplatform/social/rest/invitation/PeopleAndSpacesRestService.java
* https://github.com/Meeds-io/social/blob/develop/component/service/src/main/java/org/exoplatform/social/rest/suggest/PeopleRestServices.java